magicblock-aperture/src/server/http/mod.rs

```diff
@@ -104,6 +104,7 @@ impl HttpServer {
             let builder = conn::auto::Builder::new(TokioExecutor::new());
             let connection = builder.serve_connection(io, handler);
             tokio::pin!(connection);
+            let mut terminating = false;
```

Comment:
Nit: I wonder if there is a better way than using this flag here. Maybe putting the connection inside an `Option` and then doing something like `if let Some(..)`?